iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java

```diff
@@ -709,180 +693,182 @@ public boolean appendFileToFileNode(String fileNodeName, IntervalFileNode append
+      }
+      String fileNodePath = TsFileDBConf.fileNodeDir;
+      fileNodePath = standardizeDir(fileNodePath) + processorName;
+      FileUtils.deleteDirectory(new File(fileNodePath));
+
+      cleanBufferWrite(processorName);
+      cleanBufferWrite(processorName);
+
+      MultiFileLogNodeManager.getInstance()
+              .deleteNode(processorName + IoTDBConstant.BUFFERWRITE_LOG_NODE_SUFFIX);
+      MultiFileLogNodeManager.getInstance()
+              .deleteNode(processorName + IoTDBConstant.OVERFLOW_LOG_NODE_SUFFIX);
+    } catch (IOException e) {
+      LOGGER.error("Delete the filenode processor {} error.", processorName, e);
+      throw new FileNodeManagerException(e);
+    } finally {
+      fileNodeManagerStatus = FileNodeManagerStatus.NONE;
+    }
+  }
+
+  private void cleanBufferWrite(String processorName) throws IOException {
+    List<String> bufferwritePathList = directories.getAllTsFileFolders();
+    for (String bufferwritePath : bufferwritePathList) {
+      bufferwritePath = standardizeDir(bufferwritePath) + processorName;
+      File bufferDir = new File(bufferwritePath);
+      // free and close the streams under this bufferwrite directory
+      if (!bufferDir.exists())
+        continue;
+      File[] bufferFiles = bufferDir.listFiles();
+      if (bufferFiles != null) {
+        for (File bufferFile : bufferFiles) {
+          FileReaderManager.getInstance().closeFileAndRemoveReader(bufferFile.getPath());
+        }
+      }
+      FileUtils.deleteDirectory(new File(bufferwritePath));
+    }
+  }
+
+  private void deleteFileNodeBlocked(String processorName) throws FileNodeManagerException {
+    LOGGER.info("Forced to delete the filenode processor {}", processorName);
+    FileNodeProcessor processor = processorMap.get(processorName);
+    while (true) {
       if (processor.tryWriteLock()) {
         try {
           if (processor.canBeClosed()) {
-            processor.close();
-            return true;
+            LOGGER.info("Delete the filenode processor {}.", processorName);
+            processor.delete();
+            processorMap.remove(processorName);
+            break;
           } else {
-            return false;
+            LOGGER.info(
+                    "Can't delete the filenode processor {}, "
+                            + "because the filenode processor can't be closed."
+                            + " Wait 100ms to retry");
           }
         } catch (ProcessorException e) {
-          LOGGER.error("Close the filenode processor {} error.", processorName, e);
+          LOGGER.error("Delete the filenode processor {} error.", processorName, e);
           throw new FileNodeManagerException(e);
         } finally {
           processor.writeUnlock();
         }
       } else {
-        return false;
+        LOGGER.info(
+                "Can't delete the filenode processor {}, because it can't get the write lock."
+                        + " Wait 100ms to retry");
       }
```

Comment:
Add the processor name, {... , processorName}

Reply:
fixed